spec/API_specification/array_object.md

```diff
@@ -727,23 +727,36 @@ Element-wise results must equal the results returned by the equivalent element-w
 (method-__matmul__)=
 ### \_\_matmul\_\_(x1, x2, /)
 
-_TODO: awaiting `matmul` functional equivalent._
+Computes the matrix product.
+
+```{note}
+
+The `matmul` function must implement the same semantics as the built-in `@` operator (see [PEP 465](https://www.python.org/dev/peps/pep-0465)).
+```
 
 #### Parameters
 
 -   **x1**: _&lt;array&gt;_
 
-    -   array instance.
+    -   array instance. Should have a numeric data type. If `x1` is one-dimensional having shape `(M)` and `x2` has more than one dimension, `x1` must be promoted to a two-dimensional array by prepending `1` to its dimensions (i.e., must have shape `(1, M)`). After matrix multiplication, the prepended dimensions in the returned array must be removed. If `x1` has more than one dimension (including after vector-to-matrix promotion), `x1` must be compatible with `x2` (see {ref}`broadcasting`). If `x1` has shape `(..., M, K)`, the innermost two dimensions form matrices on which to perform matrix multiplication. 
```

Comment:
Can we just refer to the `matmul` documentation, rather than duplicating all this content?

Reply:
We can; however, to be consistent, we'd need to do that for all other dunder methods. Personally, I am fine with duplication, as it requires readers to take fewer hops to understand the semantics for dunder APIs.